### PR TITLE
edk2 cross fix

### PR DESCRIPTION
**Pull request by avnik**
## Description of changes

Fix cross-compilation to aarch64-linux.
Closing #245188

(Candidate to backport to 23.11 as well)
@RaitoBezarius @mikatammi 
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/compilers/edk2/default.nix [hunk 1]
```diff
@@ -61,6 +61,7 @@ edk2 = stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ pythonEnv ];
   depsBuildBuild = [ buildPackages.stdenv.cc buildPackages.util-linux buildPackages.bash ];
+  buildInputs = [ libuuid ];
```
Comment (LunNova) [inline pkgs/development/compilers/edk2/default.nix:64]: libuuid is aliased to util-linuxMinimal.
Do we actually need both util-linux going into depsBuildBuild and libuuid here?

**Comment by LunNova** [comment]
@ofborg eval pkgsCross.aarch64-multiplatform.edk2

**Comment by LunNova** [comment]
@ofborg build pkgsCross.aarch64-multiplatform.edk2

File: pkgs/development/compilers/edk2/default.nix (hunk 1 above)
Comment (avnik) [reply pkgs/development/compilers/edk2/default.nix:64]: @LunNova during cross build it fail on missing `-luuid` (see linked issue https://github.com/NixOS/nixpkgs/issues/245188)

**Comment by avnik** [comment]
@LunNova I added one more commit, which fixing following OVMF issue:
```
OVMF-aarch64-unknown-linux-gnu> /build/edk2-unvendored-src/BaseTools/BinWrappers/PosixLike/VfrCompile: line 18: /build/edk2-unvendored-src/BaseTools/Source/C/bin/VfrCompile: cannot execute binary file: Exec format error
OVMF-aarch64-unknown-linux-gnu> /build/edk2-unvendored-src/BaseTools/BinWrappers/PosixLike/VfrCompile: line 18: /build/edk2-unvendored-src/BaseTools/Source/C/bin/VfrCompile: Success
OVMF-aarch64-unknown-linux-gnu> /build/edk2-unvendored-src/BaseTools/BinWrappers/PosixLike/VfrCompile: line 18: /build/edk2-unvendored-src/BaseTools/Source/C/bin/VfrCompile: cannot execute binary file: Exec format error
OVMF-aarch64-unknown-linux-gnu> /build/edk2-unvendored-src/BaseTools/BinWrappers/PosixLike/VfrCompile: line 18: /build/edk2-unvendored-src/BaseTools/Source/C/bin/VfrCompile: Success
OVMF-aarch64-unknown-linux-gnu> make: *** [GNUmakefile:300: /build/edk2-unvendored-src/Build/ArmVirtQemu-AARCH64/RELEASE_GCC5/AARCH64/MdeModulePkg/Library/FileExplorerLib/FileExplorerLib/DEBUG/FileExplorerVfr.c] Error 126
OVMF-aarch64-unknown-linux-gnu> /build/edk2-unvendored-src/BaseTools/BinWrappers/PosixLike/VfrCompile: line 18: /build/edk2-unvendored-src/BaseTools/Source/C/bin/VfrCompile: cannot execute binary file: Exec format error
OVMF-aarch64-unknown-linux-gnu> /build/edk2-unvendored-src/BaseTools/BinWrappers/PosixLike/VfrCompile: line 18: /build/edk2-unvendored-src/BaseTools/Source/C/bin/VfrCompile: Success
OVMF-aarch64-unknown-linux-gnu> make: *** [GNUmakefile:300: /build/edk2-unvendored-src/Build/ArmVirtQemu-AARCH64/RELEASE_GCC5/AARCH64/MdeModulePkg/Library/DeviceManagerUiLib/DeviceManagerUiLib/DEBUG/DeviceManagerVfr.c] Error 126
OVMF-aarch64-unknown-linux-gnu> /build/edk2-unvendored-src/BaseTools/BinWrappers/PosixLike/VfrCompile: line 18: /build/edk2-unvendored-src/BaseTools/Source/C/bin/VfrCompile: cannot execute binary file: Exec format error
OVMF-aarch64-unknown-linux-gnu> /build/edk2-unvendored-src/BaseTools/BinWrappers/PosixLike/VfrCompile: line 18: /build/edk2-unvendored-src/BaseTools/Source/C/bin/VfrCompile: Success
OVMF-aarch64-unknown-linux-gnu> make: *** [GNUmakefile:300: /build/edk2-unvendored-src/Build/ArmVirtQemu-AARCH64/RELEASE_GCC5/AARCH64/MdeModulePkg/Library/BootManagerUiLib/BootManagerUiLib/DEBUG/BootManagerVfr.c] Error 126
```

But I not sure yet, if resulting OVMF binaries works.

**Comment by RaitoBezarius** [comment]
This is an incorrect fix.

**Comment by adamcstephens** [comment]
This should be fixed by #270026

**Pull request closed without merging**